### ISSUE TITLE
Checkpoint Library Management 

### ISSUE THREAD

**Issue by Phantoms46** [NitrOP7674/FreeplayCheckpoint#15]
Dear NitrOP7674,

First of all I absolutely love your Freeplay Checkpoint plugin! Thank you so much for creating this amazing training tool.

I am writing to you because I ran into an issue and wanted to run it by you. It turns out I might love your plugin a bit too much, because I am now up to over 500 checkpoints (I already deleted all my checkpoints after getting over 200+ checkpoints a couple times). 

Ideally I would want to have a better way to manage my library of checkpoints. My initial idea was something like categories or “sub-folders” of some sort. For example:
Sub-folder 1: power shots
Sub-folder 2: wall to air dribble
Sub-folder 3: preflip-air dribble.
…etc.
and then having the option to cycle through checkpoints in a given subfolder (e.g. let’s say I want to practice power shots, I can do power shot 1, 2, 3, etc.)
but also the option to cycle through subfolders.
So after cycling through all my checkpoints in a given subfolder, change subfolders.
For example I practice power shots for a couple of minutes, then I want to change to preflip air dribbles, then I can just change subfolder.

I hope what I am trying to explain makes sense. If not, I would be happy to clarify.

I tried doing this manually by going in the plugins folder and opening the “checkpointplugin.xyz” files, but it appears that the library of checkpoints is encoded.

Do you think this is something that could be done? It would help me tremendously in my training.

Thank you very much in advance,

Phantoms

**Comment by NitrOP7674** [comment]
How about a new command, `cpt_filename <fn>`, where `<fn>` would be the filename used for saving and loading checkpoints?  You could, e.g., bind `cpt_filename powershots.cpt` to one function key, `cpt_filename w2adribble.cpt` to another, etc.  Or use `F6` to pull up the console and enter those commands manually.

Would that be useful?  It should be pretty easy to implement on my end (which is ultimately what I'm looking for since I don't have all that much time to spend on developing this lately).

**Comment by Phantoms46** [comment]
Yes, that new command would be very useful. Thank you so much!

**Comment by NitrOP7674** [comment]
If you wouldn't mind trying out a pre-release version with this feature, I'd appreciate it:

https://github.com/NitrOP7674/FreeplayCheckpoint/releases/download/v1.5.0-pre/CheckpointPlugin-v1.5.0-pre.zip

**Comment by Phantoms46** [comment]
Of course! 

If I understand correctly, this is what happens:

I enter the new command cpt_filename <fn> and it starts a new "library" of checkpoints under that name.

for example I enter the command cpt_filename powershots.cpt
--> I am now in a new library with no checkpoints
----> I can save a couple interesting checkpoints (lets say 5 power shots)

I enter the new command cpt_filename DoubleTaps.cpt
--> again I am now in a new library with 0 checkpoints
----> I save a couple interesting double tap checkpoints (lets say 10 checkpoints)

If I want, I can now enter cpt_filename powershots.cpt to go back to my 5 powershots checkpoints library
and from there, I can enter cpt_filename DoubleTaps.cpt to go back to my double tap library

they are separate libraries. from the powershots.cpt library, i only have access to those 5 shots. If I want to access the 10 double tap checkpoints, I have to enter the command. I never have access to the full 15 shots at once.

is that working as intended?

the only problem I had is that when I entered the cpt_filename <fn> command intially, it started a new library, so I don't have access to my 500+ shot library anymore. Is there a default parameter I can enter to load the starting "library"? 

e.g. cpt_filename default.cpt 
--> which would load the library of 500 shots i have initially when launching RL for the first time

I hope I expressed myself clearly. Let me know if you need me to rephrase anything or have any questions.

Thank you again for your help. I can see how this will help me manage my shots in a very efficient way compared to the current release.

**Comment by NitrOP7674** [comment]
Yes, that all sounds right. The default file for checkpoints is `freeplaycheckpoint.data`.

Hmm, I'm not sure how best to transfer shots from one file to another. I suppose `cpt_copy` and `cpt_paste` would be the best way. Maybe copy all the existing shots, organize them out of game, then transfer them back in by pasting after you create a new file. Not ideal but probably the easiest.

**Comment by Phantoms46** [comment]
Alright, cool, typing 'cpt_filename freeplaycheckpoint.data' in console did indeed load my original 500+ shot library.

And we had the same idea! I already exported all my shots to excel, along with the necessary string (using cpt_copy) to load them back up (while I was originally waiting to hear from you and started working on a way to find a solution I could come up with on my end).

With your new command and my data, it looks like I will be able to organize my library exactly like I envisioned intially 😃
